app/lib/frontend/handlers/account.dart

```diff
@@ -42,12 +42,25 @@ Future<shelf.Response> updateSessionHandler(shelf.Request request,
 
   final newSession = await accountBackend.createNewSession(
       imageUrl: clientSessionData.imageUrl);
-  final cookie = Cookie(pubSessionCookieName, newSession.sessionId)
-    ..expires = newSession.expires
-    ..httpOnly = true
-    ..path = '/';
   final headers = <String, String>{
-    HttpHeaders.setCookieHeader: cookie.toString(),
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+    HttpHeaders.setCookieHeader: [
+      // Cookies prefixed '__Host-' must:
+      //  * be set by a HTTPS response,
+      //  * not feature a 'Domain' directive, and,
+      //  * have 'Path=/' directive.
+      // Hence, such a cookie cannot have been set by another website or an
+      // HTTP proxy for this website.
+      '$pubSessionCookieName=${newSession.sessionId}',
+      // Send cookie to anything under '/' required by '__Host-' prefix.
+      'Path=/',
+      // Do not include the cookie in CORS requests, unless the request is a
+      // top-level navigation to the site, as recommended in:
+      // https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-02#section-8.8.2
+      'SameSite=Lax',
+      'Secure', // Only allow this cookie to be sent when making HTTPS requests.
```

Comment:
can we have a conditional for localhost?
e.g. 
```
if (request.requestedUri.scheme != 'http') 'Secure',
```
(or even `if (!request.requestedUri.toString().startsWith('http://localhost:8080/') 'Secure',`

Reply:
I think we should somehow put into the `configuration` whether the process was started with `--local-dev` and only in that case should we do something special.

People might be hitting the service with `Host: localhost` in production, you never know. Certainly there is no good reason to allow things like that in production. Let's be defensive :)

Reply:
++